webui/react/src/stores/users.tsx

```diff
@@ -2,116 +2,124 @@ import { Map } from 'immutable';
       })
       .catch((e) => handleError(e, { publicSubject: 'Unable to fetch current user.' }));
-  };
-
-  public updateCurrentUser = (id: number | null) => {
-    if (id === null) this.#currentUserId.set(NotLoaded);
-    else this.#currentUserId.set(Loaded(id));
-  };
-
-  public getUsers = (cfg?: FetchUsersConfig): Observable<Loadable<DetailedUserList>> => {
-    const config = cfg ?? {};
-
-    return Observable.select([this.#usersByKey, this.#users], (map, users) => {
-      const usersPagination = map.get(encodeParams(config));
 
-      if (!usersPagination) return NotLoaded;
+    return () => canceler.abort();
+  }
 
-      const userPage: DetailedUserList = {
-        pagination: usersPagination.pagination,
-        users: usersPagination.users.flatMap((userId) => {
-          const user = users.get(userId);
+  public fetchUsers(params: GetUsersParams = {}, signal?: AbortSignal) {
+    const canceler = new AbortController();
 
-          return user ? [user] : [];
-        }),
-      };
-
-      return Loaded(userPage);
-    });
-  };
-
-  public ensureUsersFetched = (
-    canceler: AbortController,
-    cfg?: FetchUsersConfig,
-    refresh = false,
-  ) => {
-    const config = cfg ?? {};
-    const usersPagination = this.#usersByKey.get().get(encodeParams(config));
-
-    if (!refresh && usersPagination) return;
-
-    getUsers(config, { signal: canceler?.signal })
+    getUsers(params, { signal: signal ?? canceler.signal })
       .then((response) => {
-        this.updateUsersByKey(config, response);
-        this.updateUsers(response.users);
+        this.#usersBySearch.update((map) =>
+          map.set(encodeParams(params), {
+            pagination: response.pagination,
+            userIds: response.users.map((user) => user.id),
+          }),
+        );
+        this.#usersById.update((prev) =>
+          prev.withMutations((map) => {
+            response.users.forEach((user) => map.set(user.id, user));
+          }),
+        );
       })
       .catch((e) => handleError(e, { publicSubject: 'Unable to fetch users.' }));
-  };
 
-  public updateUsers = (users: DetailedUser | DetailedUser[]) => {
-    this.#users.update((map) => {
-      return map.withMutations((map) => {
-        if (Array.isArray(users)) users.forEach((user) => map.set(user.id, user));
-        else map.set(users.id, users);
-      });
-    });
-  };
-
-  private updateUsersByKey = (
-    config: FetchUsersConfig | Record<string, never>,
-    usersList: DetailedUserList,
-  ) => {
-    const usersPages = {
-      pagination: usersList.pagination,
-      users: usersList.users.map((user) => user.id),
-    };
-
-    this.#usersByKey.update((map) => map.set(encodeParams(config), usersPages));
-  };
+    return () => canceler.abort();
+  }
+
+  public reset() {
+    this.#currentUser.set(NotLoaded);
+    this.#usersById.set(Map());
+    this.#usersBySearch.set(Map());
+  }
+
+  protected async poll(params: GetUsersParams = {}) {
```

Comment:
Why are we polling for users?

Reply:
user polling already exists in our codebase (via `usePolling`), so the changes were moving those polling into the user store. The discussion of whether those polling needs to continue to exist can be had but wasn't taken into consideration for this area.